pkg/volume/flexvolume/attacher.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flexvolume
+
+import (
+	"path"
+	"time"
+
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+type flexVolumeAttacher struct {
+	plugin *flexVolumePlugin
+}
+
+var _ volume.Attacher = &flexVolumeAttacher{}
+
+// Attach is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) Attach(spec *volume.Spec, hostName string) (string, error) {
+	call := a.plugin.NewDriverCall(attachCmd)
+	call.AppendSpec(spec, a.plugin.host, nil)
+	call.Append(hostName)
+
+	status, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		return (*attacherDefaults)(a).Attach(spec, hostName)
+	} else if err != nil {
+		return "", err
+	}
+	return status.Device, err
+}
+
+// WaitForAttach is part of the volume.Attacher interface
+func (a *flexVolumeAttacher) WaitForAttach(spec *volume.Spec, devicePath string, timeout time.Duration) (string, error) {
+	call := a.plugin.NewDriverCallWithTimeout(waitForAttachCmd, timeout)
+	call.AppendSpec(spec, a.plugin.host, nil)
+
+	status, err := call.Run()
+	if isCmdNotSupportedErr(err) {
+		return (*attacherDefaults)(a).WaitForAttach(spec, devicePath, timeout)
+	} else if err != nil {
+		return "", err
+	}
+	return status.Device, nil
```

Comment:
I got confused by the name. I thought `status.Device` was the name of a device. But it appears it contains the path. Maybe rename it to `DevicePath` to be explicit.